crates/vault-registry/src/lib.rs

```diff
@@ -245,6 +246,7 @@ pub mod pallet {
         /// * `InsufficientCollateralAvailable` - if the vault does not own enough collateral
         #[pallet::weight(<T as Config>::WeightInfo::withdraw_collateral())]
         #[transactional]
+/// SBPM1 #[transactional] is consider alpha, use carefully. Also it introduces a cost so it might make sense to try to avoid its usage.
```

Comment:
The alternative is equally complex in terms of time as we need to be very confident that no invalid state transitions are made. We are not opposed to spending time on this and we anyway plan to introduce more formal verification which should highlight such edge-cases. However, I firmly believe `transactional` should be enabled by default to protect naive implementations from exploitation.